api/routes/generate.js

```diff
@@ -10,35 +10,30 @@ var thunkify = require('thunkify');
 // CRX packaging module, instantiated with the `new` keyword
 var Extension = require('crx');
 
+var knownHosts = {
```

Comment:
Its much easier to maintain when you have different functions for getting the icon and the title, and they are not tangled in one code block. It's completelly coincidental that both functions use the same hostname as exceptional case.